### PR TITLE
update: Drop plugins for SwiftPM

### LINKED ISSUE #2565: Drop support for SPM plug-ins
~`--disable-sandbox` should be the preferred command line option in the docs because it enables the network requests needed for `fetch-schema`.~

Currently the SwiftPM plug-in environment is a nice-to-have but suffers from some limitations, such as the Xcode experience and needing `--disable-sandbox` for extended functionality. We're making a decision to drop support for SwiftPM plug-ins and bring them back once the environment is more mature to fully support the experience we want to deliver.

### PR DESCRIPTION
**Pull request by calvincestari**
Closes #2565 

This removes the three plugins from our package definition, preventing the CLI being used as a plugin in both `swift package` commands as well as from within Xcode.

We regret making this change but the current workflow within SwiftPM and Xcode is not suitable to the experience we want to deliver. Once those plugin environments mature a bit more we can bring back the plugins, they do make version alignment easier.

### CODE REVIEW

**Review by AnthonyMDev — APPROVED** [review]

File: docs/source/code-generation/codegen-cli.mdx [hunk 1]
```diff
@@ -1,7 +1,9 @@
 ---
-title: The codegen CLI
+title: The Codegen CLI
```
[the remark this answers is not in the document]
Comment (calvincestari) [reply docs/source/code-generation/codegen-cli.mdx:2]: Actually we should match the style of other Apollo documentation, which is sentence case.

File: docs/source/code-generation/introduction.mdx [hunk 2]
```diff
@@ -50,7 +52,7 @@ The Apollo code generation engine can be run using one of two methods:
 
 **This is the recommended method for most use cases.**
 
-The [Codegen CLI](./codegen-cli) is the simplest way to get started with Apollo iOS. The correct version of the CLI is automatically bundled with our SPM and Cocoapod packages, so you never have to worry about installation or updates.
+The [Codegen CLI](./codegen-cli) is the simplest way to get started with Apollo iOS. The correct version of the CLI is automatically bundled with our Cocoapod package, so you never have to worry about installation or updates.
```
Comment (calvincestari) [inline docs/source/code-generation/introduction.mdx:55]: Why the change to remove SPM here?

File: docs/shared/spm-install-cli.mdx [hunk 3]
```diff
@@ -0,0 +1,19 @@
+The Apollo iOS SPM package includes the Codegen CLI as an executable target. This ensures you always have a valid CLI version for your Apollo iOS version.
+
+To simplify accessing the Codegen CLI, you can run the included `InstallCLI` SPM plugin.
+
+This plugin builds the CLI and creates a sybmolic link to the executable in your project root.
```
Comment (calvincestari) [inline docs/shared/spm-install-cli.mdx:5]:
```suggestion
This plugin builds the CLI and creates a symbolic link to the executable in your project root.
```

File: docs/source/code-generation/introduction.mdx (hunk 2 above)
Comment (calvincestari) [reply docs/source/code-generation/introduction.mdx:55]: This may have been my change when I pulled out the plugins and just wasn't added back when you created the new installation plugin?

**Review by calvincestari — COMMENTED** [review]
I can't approve this PR because I created it, 'Comment' is the only option.
